### PR TITLE
Support http session

### PR DESCRIPTION
**Pull request by abrisan**
Hello fellow trino-ers 👋 !

TL;DR

- Support passing in a `http_session` to `dbapi.connect`
- Refactor tests to make them a bit more pythonic

This Pull Request adds a feature to the dbapi interface offered by the library, namely it exposes the option to pass in an http session. This is particularly useful when using proxies to cross network boundaries.

We noticed this was missing as we were migrating some tasks from the old `pyhive[presto]` sqlalchemy adapter to `sqlalchemy-trino`.

I have also taken the liberty to refactor `test_client.py` and replace some of the custom components in there with offerings from the Python standard library. This should make test maintenance easier, and it should also make it easier for experienced Python 🐍 devs to make contributions.

Thanks for making trino awesome! 💯

### CODE REVIEW

**Review by ebyhr — COMMENTED** [review]
Thanks for your contribution! Could you take a look at CI failure?

**Comment by abrisan** [comment]
@ebyhr I think this might pass - can we try again please?

**Comment by abrisan** [comment]
🎉 passing now @ebyhr

**Comment by abrisan** [comment]
@ebyhr I’ve been added to the contributors list 🎉 , could you please review the PR? Thanks!

**Comment by hashhar** [comment]
@cla-bot check

**Review by hashhar — APPROVED** [review]
LGTM.

PTAL @ebyhr if you have some comments.

**Review by ebyhr — APPROVED** [review]
Could you squash the last commit "Fix CI" into the previous one?

**Comment by abrisan** [comment]
Thanks for the review @ebyhr ! Fixed up the commit :)

**Comment by ebyhr** [comment]
Merged, thanks!
